service/src/main/java/uk/gov/hmcts/reform/fpl/utils/SecureDocStoreHelper.java

```diff
@@ -0,0 +1,189 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
+import uk.gov.hmcts.reform.ccd.document.am.model.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.FeatureToggleService;
+import uk.gov.hmcts.reform.fpl.service.SecureDocStoreService;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+
+@Slf4j
+public class SecureDocStoreHelper {
+
+    private FeatureToggleService featureToggleService;
+    private SecureDocStoreService secureDocStoreService;
+
+    public SecureDocStoreHelper(SecureDocStoreService secureDocStoreService,
+                                FeatureToggleService featureToggleService) {
+        this.featureToggleService = featureToggleService;
+        this.secureDocStoreService = secureDocStoreService;
+    }
+
+    public byte[] download(final String documentUrlString) {
+        return download(documentUrlString, null);
+    }
+
+    /**
+     * If secure doc store toggle is off, attempt to fire the API and log any exceptions caught.
+     * @param documentUrlString  document url in string
+     * @param oldDmStoreApproach if featureToggleService.isSecureDocstoreEnabled() = false, it is a mandatory parameter.
+     *                           otherwise, UnsupportedOperationException will be thrown.
+     * @return byte array of the file
+     */
+    @SneakyThrows
+    public byte[] download(final String documentUrlString, Callable<byte[]> oldDmStoreApproach) {
+        try {
+            log.info("Downloading document: {}", documentUrlString);
+            byte[] bytesFromSecureDocStore = secureDocStoreService.downloadDocument(documentUrlString);
+            if (featureToggleService.isSecureDocstoreEnabled()) {
+                return bytesFromSecureDocStore;
+            } else {
+                log.info("Downloaded document attempted from CDAM without error: {}", documentUrlString);
+            }
```

Comment:
I think for testing purposes we can keep these 'success logs' for now, but I think we should remove them before we deploy, just to save on unnecessary logging.

Reply:
(I won't add the comment to upload/download metadata but same applies across all 3, can save a little money (based on how much we download docs, might result in a lot of logs!)